src/lib.rs

```diff
@@ -25,3 +35,88 @@ pub fn minor_version(deploy_version: &str) -> Option<u32> {
         .map(|v| v.parse::<u32>().ok())
         .unwrap_or(None)
 }
+
+/// Load the `kube::Configuration` giving priority to local, falling back to in-cluster config
+pub async fn load_kube_config() -> Configuration {
+    config::load_kube_config()
+        .await
+        .unwrap_or_else(|_| config::incluster_config().expect("Failed to get local kube config and incluster config"))
+}
+
+#[derive(Clone)]
+pub struct Controller {
+    client: APIClient,
+    namespace: String,
+    gordo_rf: Reflector<Gordo>,
+    model_rf: Reflector<Model>,
+}
+
+impl Controller {
+    /// Create a new instance of the Gordo Controller
+    pub async fn new(kube_config: Configuration) -> Self {
+        let namespace = kube_config.default_ns.to_owned();
+        let client = APIClient::new(kube_config);
+
+        let model_resource = load_model_resource(&client, &namespace);
+        let model_rf = Reflector::new(model_resource.clone()).timeout(15).init().await.unwrap();
```

Comment:
What happends when it timeouts, do we know?

Reply:
Limits the [duration of the call](https://docs.rs/kube/0.20.0/kube/api/struct.Reflector.html#method.timeout) I see in the latest release changes it to 300 seconds but guess we want more frequent updates than that?

Reply:
I would assume that k8s streams the updates over the live connection, so it is not such that we only get updates every 15 (or 300) second (?). The way I read it is that it forcefully makes a new connection every *timeout* seconds.

Reply:
That's also how I understand it. It appears in the later release (after they handle the `Stream` not being `Send`) the default will be 300 seconds which implies, as you say, will stream results during the open connection. Where now it's a balance of how long blocking is acceptable before the actual state will be updated. :thinking: